src/services/completions.ts

```diff
@@ -2179,7 +2185,13 @@ namespace ts.Completions {
             case SyntaxKind.CommaToken: // class c {getValue(): number, | }
                 return tryCast(contextToken.parent, isObjectTypeDeclaration);
             default:
-                if (!isFromObjectTypeDeclaration(contextToken)) return undefined;
+                if (!isFromObjectTypeDeclaration(contextToken)) {
+                    // class c extends React.Component { a: () => 1\n| }
+                    if (getLineAndCharacterOfPosition(sourceFile, contextToken.getEnd()).line !== getLineAndCharacterOfPosition(sourceFile, position).line && isObjectTypeDeclaration(location)) {

```

Comment:
Can you please reverse the order of checks here (that is checking `isObjectTypeDeclaration(location)` before line?

Reply:
Switching the if ordering changes the behavior from what it previously was, and starts to trigger behavior which I think we don't want.

[This fourslash](https://github.com/Microsoft/TypeScript/blob/e55f97ec282d720f9bf58792ece0acc048202f93/tests/cases/fourslash/completionsGeneratorFunctions.ts)  test fails:
```
/// <reference path="fourslash.ts" />

////function /*a*/ ;
////function* /*b*/ ;
////interface I {
////    abstract baseMethod(): Iterable<number>;
////}
////class C implements I {
////    */*c*/ ;
////    public */*d*/
////}
////const o: I = {
////    */*e*/
////};
////1 * /*f*/

verify.completions(
    { marker: ["a", "b"], exact: undefined, isNewIdentifierLocation: true },
    { marker: ["c", "d"], exact: ["baseMethod"], isNewIdentifierLocation: true },
    { marker: "e", exact: ["baseMethod"] },
    { marker: "f", includes: [{ name: "Number", sortText: completion.SortText.GlobalsOrKeywords }] },
);
```

With 

```
  1) fourslash tests
       tests/cases/fourslash/completionsGeneratorFunctions.ts
         fourslash test completionsGeneratorFunctions.ts runs correctly:
     Error: At f: Expected 'isNewIdentifierLocation' to be false, got true
```

Which is returning an empty array of completions